File: isolate/Transferable Object/dart2js/psum.dart
import 'dart:html';
// import 'dart:typed_data';

// dart-enabled web worker does not be supported at present.
// https://github.com/dart-lang/sdk/issues/21573 
main() {
  window.onMessage.listen((e) {
    var view = e.data.asUint8List();
    var res = 0;
    for (var i in view) {
      if (i % 2 == 0) res += i;
    }
    window.postMessage(res,'*');
    window.close();
  });
}

File: isolate/Transferable Object/dart2js/psum.js
self.onmessage = function(e) {
    var view = new Uint8Array(e.data);
    var res = 0;
    for (var i = 0;i < view.length;i++) {
        if (view[i] % 2 == 0) res += view[i];
    }
    postMessage(res);
    close();
}; 

